app/src/hooks/useRelay.tsx

```diff
@@ -0,0 +1,67 @@
+import { BigNumber } from 'ethers/utils'
+import { useEffect, useState } from 'react'
+
+import { useConnectedWeb3Context } from '../contexts/connectedWeb3'
+import { RelayService } from '../services'
+import { getNativeAsset } from '../util/networks'
+import { Token } from '../util/types'
+
+interface Status {
+  active: boolean
+}
+
+interface RelayServiceResponse {
+  relayAddress: string
+  relayFee: BigNumber
+  relayFeeGreaterThanAmount: boolean
+  relayFeeGreaterThanBalance: boolean
+}
+
+export const useRelay = (amount?: BigNumber, collateral?: Token): RelayServiceResponse => {
+  const { account, balances, networkId, relay } = useConnectedWeb3Context()
+
+  const [relayAddress, setRelayAddress] = useState('')
+  const [relayFee, setRelayFee] = useState(new BigNumber('0'))
+  const [relayFeeGreaterThanAmount, setRelayFeeGreaterThanAmount] = useState(false)
+  const [relayFeeGreaterThanBalance, setRelayFeeGreaterThanBalance] = useState(false)
+
+  const fetchRelayInfo = async (status?: Status) => {
+    const relayService = new RelayService()
+    const { address, fee } = await relayService.getInfo()
+    if (!status || status.active) {
+      setRelayAddress(address)
+
+      const feeBN = new BigNumber(fee)
+      setRelayFee(feeBN)
+      setRelayFeeGreaterThanBalance(feeBN.gt(balances.xDaiBalance))
+    }
+  }
+
+  useEffect(() => {
+    const status = { active: true }
+    if (account && relay) {
+      fetchRelayInfo(status)
+    } else {
+      setRelayAddress('')
+      setRelayFee(new BigNumber('0'))
+      setRelayFeeGreaterThanBalance(false)
+    }
+    return () => {
+      status.active = false
+    }
```

Comment:
this prevents against race conditions when requesting data, e.g. imagine we start requesting data while the user is using the relay and the user switches to rinkeby mid request - it is possible that the state could be updated with data we don't want once the `relayService.getInfo()` request finishes. 

The line you commented will make sure status.active is set to false if the hook is updated, so in our scenario above once `relayService.getInfo()` is finished we check `status.active`, if it is true that means the hook has not updated since the request was made and it is safe to update the state.